packages/types/src/CryptoCallbacks.ts

```diff
@@ -6,107 +6,154 @@
  */
 
 import type { SignerPayloadJSON } from '@polkadot/types/types'
+import {
+  DidResourceUri,
+  DidUri,
+  DidVerificationKey,
+  VerificationKeyRelationship,
+} from './DidDocument.js'
 
-export type SigningAlgorithms = 'ed25519' | 'sr25519' | 'ecdsa-secp256k1'
+const signingAlgorithmsC = ['ed25519', 'sr25519', 'ecdsa-secp256k1'] as const
+export const signingAlgorithms = signingAlgorithmsC as unknown as string[]
+export type SigningAlgorithms = typeof signingAlgorithmsC[number]
 
 export type EncryptionAlgorithms = 'x25519-xsalsa20-poly1305'
 
 /**
- * Base interface for all {en/de}cryption & signing requests.
+ * Base interface for all signing requests.
  */
-export interface RequestData<
-  A extends SigningAlgorithms | EncryptionAlgorithms
-> {
+export interface SignRequestData {
   /**
-   * Identifier for the encryption/signing algorithm to use.
+   * Data to be signed.
    */
-  alg: A
+  data: Uint8Array
   /**
-   * Public key as u8a identifying the keypair to use (in combination with the alg).
+   * Info for extensions to display to user.
+   * Available when signing extrinsics.
    */
-  publicKey: Uint8Array
+  meta?: Partial<SignerPayloadJSON>
   /**
-   * Data to be {en/de}crypted or signed.
+   * The did key relationship to be used.
    */
-  data: Uint8Array
+  keyRelationship: VerificationKeyRelationship
 
-  [x: string]: unknown
+  /**
+   * The DID to be used for signing.
+   */
+  did: DidUri
 }
 
 /**
- * Base interface for responses to {en/de}cryption & signing requests.
+ * Base interface for responses to signing requests.
  */
-export interface ResponseData<
-  A extends SigningAlgorithms | EncryptionAlgorithms
-> {
+export interface SignResponseData {
   /**
-   * Identifier for the encryption/signing algorithm used.
+   * Result of the signing.
    */
-  alg: A
+  data: Uint8Array
   /**
-   * Result of the {en/de}cryption or signing.
+   * The did key uri used for signing.
    */
-  data: Uint8Array
-
-  [x: string]: unknown
+  keyUri: DidResourceUri
+  /**
+   * The did key type used for signing.
+   */
+  keyType: DidVerificationKey['type']
 }
 
 /**
- * Extends [[RequestData]] with optional metadata for providing info on the data to be signed, especially in case of signing extrinsics.
+ * A callback function to sign data.
+ */
+export type SignCallback = (
+  signData: SignRequestData
+) => Promise<SignResponseData>
+
+/**
+ * A callback function to sign data without an existing DID.
+ */
+export type SignWithoutDidCallback = (
```

Comment:
I love the interfaces, but the names are a little backwards, no? I'd expect the callback that needs a DID for signing to be called something like `DidSignCallback` and the one that does not need a DID something like `KeyPairSignCallback` or just `SignCallback`.

We also talked about moving the without Did type to live with the did stuff if it has just a single use case (did creation). Did you decide against that?

Reply:
no, it is in the todo list of this PR

Reply:
If our only signCallback interface is about DIDs, I guess it's fine if we don't explicitly mention DIDs in its name